src/sentry/static/sentry/app/views/settings/projectDataPrivacy/projectDataPrivacyRulesPanel.tsx

```diff
@@ -0,0 +1,223 @@
+    }));
+  };
+
+  handleChange = (updatedRule: Rule) => {
+    this.setState(prevState => ({
+      rules: prevState.rules.map(rule => {
+        if (rule.id === updatedRule.id) {
+          return updatedRule;
+        }
+        return rule;
+      }),
+    }));
+  };
+
+  handleSubmit = () => {
+    requestResponse = this.state.rules;
+
+    this.setState(
+      prevState => ({
+        status: 'success',
+        savedRules: prevState.rules,
+      }),
+      () => {
+        this.resetStatus();
+      }
+    );
+  };
+
+  handleValidation = () => {
+    const {rules} = this.state;
+    const isAnyRuleFieldEmpty = rules.find(rule =>
+      Object.keys(rule).find(ruleKey => !rule[ruleKey])
+    );
+
+    const isFormValid = !isAnyRuleFieldEmpty;
+
+    if (isFormValid) {
+      this.handleSubmit();
+    } else {
+      this.setState(
+        {
+          status: 'error',
+        },
+        () => {
+          this.resetStatus();
+        }
+      );
+    }
+  };
+
+  handleSaveForm = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    this.handleValidation();
+  };
+
+  handleCancelForm = () => {
+    this.setState(
+      prevState => ({
+        status: 'cancelling',
+        rules: prevState.savedRules,
+      }),
+      () => {
+        this.resetStatus();
+      }
+    );
+  };
+
+  render() {
+    const {rules, status} = this.state;
+
+    activeIndicatorMessageStatus(status);
+
+    return (
+      <React.Fragment>
+        <Panel>
+          <PanelHeader>{t('Data Privacy Rules')}</PanelHeader>
+          <PanelBody>
+            {rules.map(rule => (
+              <ProjectDataPrivacyRulesForm
+                key={rule.id}
+                onDelete={this.handleDeleteRule}
+                onChange={this.handleChange}
+                id={rule.id}
+                action={rule.action}
+                data={rule.data}
+                customRegularExpression={rule.customRegularExpression}
+                from={rule.from}
+              />
+            ))}
+            <PanelAction>
+              <StyledButton
+                icon={<IconAdd circle />}
+                onClick={this.handleAddRule}
+                borderless
+              >
+                {t('Add Rule')}
+              </StyledButton>
+            </PanelAction>
+          </PanelBody>
+        </Panel>
+        {rules.length > 0 && (
+          <StyledButtonBar gap={1.5}>
+            <Button onClick={this.handleCancelForm}>{t('Cancel')}</Button>
+            <Button priority="primary" onClick={this.handleSaveForm}>
+              {t('Save Rules')}
+            </Button>
+          </StyledButtonBar>
+        )}
+      </React.Fragment>
+    );
+  }
+}
+
+export default ProjectDataPrivacyRulesPanel;
+
+const PanelAction = styled('div')`
+  padding: ${space(2)} ${space(3)};
+`;
+
+// TODO(style): clarify if the color below should be added to the theme or if we should use another color - #3d74db
```

Comment:
Is there a color in theme that is similar to this?

Reply:
Hm... the link color is similar... I will update ;-)